eduhelx_jupyterlab_student/handlers.py

```diff
@@ -123,20 +127,60 @@ async def post(self):
         
         self.finish(json.dumps(os.path.join("/", frontend_cloned_path)))
 
-class CourseAndStudentHandler(BaseHandler):
+class PollCourseStudentHandler(BaseHandler):
     @tornado.web.authenticated
     async def get(self):
+        # If this is the first poll the client has made, the `current_value` argument will not be present.
+        # Note: we are not deserializing `current_value`; it is a JSON-serialized string.
+        current_value: str | None = None
+        if "current_value" in self.request.arguments:
+            current_value = self.get_argument("current_value")
+
+        start = time.time()
+        while (elapsed := time.time() - start) < self.config.LONG_POLLING_TIMEOUT_SECONDS:
+            new_value = await CourseAndStudentHandler.get_value(self)
+            if new_value != current_value:
+                self.finish(new_value)
+                return
+            asyncio.sleep(self.config.LONG_POLLING_SLEEP_INTERVAL_SECONDS)
```

Comment:
why do we need this polling logic? is `api.get_my_user()` and `api.get_course()` expected to take a while to respond?

Reply:
Explained over call, mostly because of a browser bug whose fix naturally led to us implementing polling here instead